### ISSUE TITLE
belongs_to in jruby 9.2 and activerecord 6.1.0 raises ArgumentError on method_defined?

### ISSUE THREAD

**Issue by keegangroth** [rails/rails#41009]
looks like a direct result of https://github.com/rails/rails/commit/d6252aedadf1c02b45afdb6d842a2f42be9305ea

### Steps to reproduce
```
# frozen_string_literal: true

require "bundler/inline"

gemfile(true) do
  source "https://rubygems.org"

  git_source(:github) { |repo| "https://github.com/#{repo}.git" }

  # Activate the gem you are reporting the issue against.
  gem "activerecord", "6.1.0"
  gem "activerecord-jdbcsqlite3-adapter", "61.0", platform: :jruby
  gem "sqlite3", platform: :mri
end

require "active_record"
require "minitest/autorun"
require "logger"

# This connection will do for database-independent bug reports.
ActiveRecord::Base.establish_connection(adapter: "sqlite3", database: ":memory:")
ActiveRecord::Base.logger = Logger.new(STDOUT)

ActiveRecord::Schema.define do
  create_table :posts, force: true do |t|
  end

  create_table :comments, force: true do |t|
    t.integer :post_id
  end
end

class Post < ActiveRecord::Base
  has_many :comments
end

class Comment < ActiveRecord::Base
  belongs_to :post
end

class BugTest < Minitest::Test
  def test_association_stuff
    post = Post.create!
    post.comments << Comment.create!

    assert_equal 1, post.comments.count
    assert_equal 1, Comment.count
    assert_equal post.id, Comment.first.post.id
  end
end
```

### Expected behavior
Using ruby 2.5.3p105 (2018-10-18 revision 65156) [x86_64-darwin17]
```
Finished in 0.012281s, 81.4266 runs/s, 244.2798 assertions/s.

1 runs, 3 assertions, 0 failures, 0 errors, 0 skips
```

### Actual behavior
```
ArgumentError: wrong number of arguments calling `method_defined?` (given 2, expected 1)
            define_non_cyclic_method at /Users/kroth/.rvm/gems/jruby-9.2.6.0/gems/activerecord-6.1.0/lib/active_record/autosave_association.rb:173
  add_autosave_association_callbacks at /Users/kroth/.rvm/gems/jruby-9.2.6.0/gems/activerecord-6.1.0/lib/active_record/autosave_association.rb:208
                               build at /Users/kroth/.rvm/gems/jruby-9.2.6.0/gems/activerecord-6.1.0/lib/active_record/autosave_association.rb:143
                    define_callbacks at /Users/kroth/.rvm/gems/jruby-9.2.6.0/gems/activerecord-6.1.0/lib/active_record/associations/builder/association.rb:82
                                each at org/jruby/RubyArray.java:1792
                    define_callbacks at /Users/kroth/.rvm/gems/jruby-9.2.6.0/gems/activerecord-6.1.0/lib/active_record/associations/builder/association.rb:81
                    define_callbacks at /Users/kroth/.rvm/gems/jruby-9.2.6.0/gems/activerecord-6.1.0/lib/active_record/associations/builder/collection_association.rb:14
                               build at /Users/kroth/.rvm/gems/jruby-9.2.6.0/gems/activerecord-6.1.0/lib/active_record/associations/builder/association.rb:34
                            has_many at /Users/kroth/.rvm/gems/jruby-9.2.6.0/gems/activerecord-6.1.0/lib/active_record/associations.rb:1456
                       <module:Post> at test.rb:33
                              <main> at test.rb:32
```

### System configuration
**Rails version**:
activerecord 6.1.0

**Ruby version**:
$ ruby -v
jruby 9.2.6.0 (2.5.3) 2019-02-11 15ba00b Java HotSpot(TM) 64-Bit Server VM 25.192-b12 on 1.8.0_192-b12 +jit [darwin-x86_64]

**Comment by rafaelfranca** [comment]
Thank you for the issue. `method_defined?` in Ruby takes 2 arguments since Ruby 2.5. This seems to be a bug in JRuby. Please report there.

**Comment by keegangroth** [comment]
not disagreeing, but jruby does report its arity "correctly"
```
jruby-9.2.6.0 :001 > Module.method(:method_defined?).arity
 => 1
```

**Comment by rafaelfranca** [comment]
Right. Then Rails does the right thing, add a refinement to Module to change the arity to 2. I believe refinements in JRuby is broken.

**Comment by keegangroth** [comment]
got it, thanks. seems to have already been fixed in jruby 9.2.12 one way or another
